app/repository/OrderRepository.scala

```diff
@@ -2,14 +2,36 @@ package repository
 
 import com.google.inject.Inject
 import dao.util.ShippearDAO
-import model.internal.OrderState.CANCELLED
+import model.internal.OrderState.{CANCELLED, PENDING_PICKUP}
+import model.internal.UserType._
 import model.internal.{Order, User}
 import service.Exception.NotFoundException
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.Success
 
 class OrderRepository @Inject()(userRepository: UserRepository)(implicit ec: ExecutionContext) extends ShippearRepository[Order] {
+  def assignCarrier(orderId : String, carrierId : String, qrCode : Array[Byte]): Future[Unit] ={
+    for{
+      order <- super.findOneById(orderId)
+      newOrder = order.copy(carrierId = Some(carrierId), qrCode = Some(qrCode), state = PENDING_PICKUP)
+      _ <- update(newOrder)
+      _ <- userRepository.updateUserOrder(newOrder.applicantId, newOrder)
+      _ <- userRepository.updateUserOrder(newOrder.participantId, newOrder)
+      _ <- userRepository.updateUserOrder(carrierId, newOrder)
+    } yield newOrder
+  }
+
+  def validateQrCode(orderId : String, userId : String, userType : UserType): Future[Any] ={
+    val order = findOneById(orderId)
+    order.map{ order =>
+        userType match{
+        case APPLICANT => order.applicantId.equals(userId)
+        case PARTICIPANT => order.participantId.equals(userId)
+        case CARRIER =>  order.carrierId.getOrElse().equals(userId)
```

Comment:
Aca habria que pensar qué pasaría si me piden validar el QR de un CARRIER, pero ese carrier no esta en la orden.

Imagino que eso deberia ser un error, ya que estamos validando un qr a una orden que no tiene un carrier asignado.

Te parece agregar en el `getOrElse(throw NotFoundException("Carrier not found")` o algo parecido para que sepamos que es un error?

Reply:
Me parece bien